opm/simulators/linalg/bda/openclKernels.cpp

```diff
@@ -395,6 +507,27 @@ void OpenclKernels::apply_stdwells_no_reorder(cl::Buffer& d_Cnnzs_ocl, cl::Buffe
         )";
     }
 
+    // multiply vector with another vector, element-wise
+    std::string OpenclKernels::get_vmul_string() {
+        return R"(
+        __kernel void vmul(
+            const double alpha,
+            __global double const *in1,
+            __global double const *in2,
+            __global double *out,
+            const int N)
+        {
+            unsigned int NUM_THREADS = get_global_size(0);
+            int idx = get_global_id(0);
+
+            while(idx < N){
+                out[idx] += alpha * in1[idx] * in2[idx];
+                idx += NUM_THREADS;
```

Comment:
The comment was outdated, I copied the comment from the header now.
```
/// multiply vector with another vector and a scalar, element-wise
/// add result to a third vector
```
This explains what the method does, although it could be hard to read.

Reply:
Thanks